tests/Valit.Tests/Collection/Primitive_Property_Collection_Validation_Tests.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Shouldly;
+using Xunit;
+
+namespace Valit.Tests.Collection
+{
+    public class Primitive_Property_Collection_Validation_Tests
+    {
+        
+        [Fact]
+        public void EnsureFor_Throws_When_Null_RuleFunc_Is_Given()
+        {
+            var exception = Record.Exception(() => {
+                ValitRules<Model>
+                    .Create()
+                    .EnsureFor(m => m.InvalidValuesCollection, (Func<IValitRule<Model,string>,IValitRule<Model,string>>)null);
+            });
+            
+            exception.ShouldBeOfType(typeof(ValitException));
+        }
+
+        [Fact]
+        public void Validation_With_Complete_Strategy_Fails_When_Invalid_Collection_Is_Given()
+        {
+            var result = ValitRules<Model>
+                .Create()
+                .EnsureFor(m => m.InvalidValuesCollection, _=>_
+                    .Required()
+                        .WithMessage("One")
+                    .Email()
+                        .WithMessage("Two"))
+                .For(_model)
+                .Validate();
+            
+            Assert.False(result.Succeeded);
+            result.ErrorMessages.Where(m => m == "One").Count().ShouldBe(1);
+            result.ErrorMessages.Where(m => m == "Two").Count().ShouldBe(3);
+        }
+
+        [Fact]
+        public void Validation_With_FailFast_Strategy_Fails_When_Invalid_Collection_Is_Given()
+        {
+            var result = ValitRules<Model>
+                .Create()
+                .WithStrategy(picker => picker.FailFast)
+                .EnsureFor(m => m.InvalidValuesCollection, _=>_
+                    .Required()
+                        .WithMessage("One")
+                    .Email()
+                        .WithMessage("Two"))
+                .For(_model)
+                .Validate();
+            
+            Assert.False(result.Succeeded);
+            result.ErrorMessages.Count().ShouldBe(1);
+            result.ErrorMessages.First().ShouldBe("Two");
```

Comment:
Since there is only one item, maybe `.Single()` is prefered?

Reply:
In this particular case there's no difference. Single thorws when sequence contains more than one element (not possible here), first thorws if nothing is returned (also not possible). I use Single over First mostly when I'm sure that I'm usin criteria on unique properties (like Id). In this case there's no predicate so both will do exactly the same thing.

Reply:
It does the same thing indeed, I was just asking it for readability.